connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/BatchExecutableProcessor.java

```diff
@@ -218,4 +219,91 @@ public void deactivateBatch(List<RegisteredExecutable> executables) {
       }
     }
   }
+
+  public CompletableFuture<RegisteredExecutable> restartFromContext(
+      RegisteredExecutable registeredExecutable, ConnectorRetryException retryException) {
+    CompletableFuture<RegisteredExecutable> future = new CompletableFuture<>();
+    LOG.warn(
+        "Inbound connector executable has requested its reactivation in {} {}",
+        retryException.getBackoffDuration().getSeconds(),
+        TimeUnit.SECONDS);
+    if (registeredExecutable instanceof RegisteredExecutable.Cancelled cancelled) {
+      InboundConnectorExecutable<InboundConnectorContext> newExecutable =
+          connectorFactory.getInstance(cancelled.context().getDefinition().type());
+      this.reactivationScheduler.schedule(
+          () ->
+              tryRestart(
+                      newExecutable,
+                      cancelled.context(),
+                      retryException.getBackoffDuration(),
+                      retryException.getRetries())
+                  .thenAccept(future::complete)
+                  .exceptionally(
+                      throwable -> {
+                        future.completeExceptionally(throwable);
+                        return null;
+                      }),
+          retryException.getBackoffDuration().getSeconds(),
+          TimeUnit.SECONDS);
+    } else {
+      LOG.error(
+          "Attempted to reactivate an inbound connector executable that is not in the cancelled state");
+      future.completeExceptionally(
+          new RuntimeException(
+              "Attempted to reactivate an inbound connector executable that is not in the cancelled state"));
+    }
+    return future;
+  }
+
+  private CompletableFuture<Activated> tryRestart(
+      InboundConnectorExecutable<InboundConnectorContext> executable,
+      InboundConnectorReportingContext context,
+      Duration delay,
+      Integer retryAttempts) {
+    CompletableFuture<Activated> future = new CompletableFuture<>();
+    try {
+      executable.activate(context);
+      LOG.info("Activation successful for {}", context.getDefinition().type());
+      future.complete(new Activated(executable, context));
+    } catch (Exception e) {
+      if (retryAttempts == 0) future.completeExceptionally(new RuntimeException("No more retries"));
+      else {
+        LOG.error(
+            "Activation failed for connector: {}. Retrying... {} retries left",
+            context.getDefinition().type(),
+            retryAttempts - 1);
+        this.reactivationScheduler.schedule(
+            () ->
+                tryRestart(executable, context, delay, retryAttempts - 1)
+                    .thenAccept(future::complete)
+                    .exceptionally(
+                        throwable -> {
+                          future.completeExceptionally(throwable);
+                          return null;
+                        }),
+            delay.getSeconds(),
+            TimeUnit.SECONDS);
+      }
+    }
+    return future;
```

Comment:
What about using https://failsafe.dev/#getting-started which we already use in other places: https://github.com/camunda/connectors/blob/725dc7e61167f16aa676edcb3de3d264ac5b8dd9/connectors/kafka/src/main/java/io/camunda/connector/kafka/inbound/KafkaConnectorConsumer.java#L117